pyiron/dft/job/generic.py

```diff
@@ -173,22 +218,21 @@ def get_kpoints(self):
 
     def get_k_mesh_by_cell(self, kpoints_per_reciprocal_angstrom, cell=None):
         """
-            get k-mesh density according to the box size.
+        Get k-mesh density according to the box size.
+
+        Args:
+            kpoints_per_reciprocal_angstrom (float): Number of k-points per reciprocal angstrom
+            cell (numpy.ndarray/list): The cell shape
+
+        Returns:
+            list/numpy.ndarray: Mesh size
 
-            Args:
-                kpoints_per_reciprocal_angstrom: (float) number of k-points per reciprocal angstrom (i.e. per 2*pi / box_length)
-                cell: (list/ndarray) 3x3 cell. If not set, the current cell is used.
         """
         if cell is None:
             if self.structure is None:
-                raise AssertionError('structure not set')
+                raise ValueError("Can't genreate k-points without structure being set")
```

Comment:
Maybe `if cell is None and self.structure is None` (saves one line :D)?

And `genreate` -> `generate`

Reply:
Typo fixed but we do need two if statements since we might have to overwrite the cell parameter.

Reply:
Ah you're right